### DISCUSSION TITLE
Convert JPG with clipping path fails on new server

### DISCUSSION THREAD

**Discussion by nikolajhave** [ImageMagick/ImageMagick#6758] [Help] [answered]
### ImageMagick version

6.9.10-23

### Operating system, version and so on

Ubuntu 20.04.6 LTS

### Description

Moving from an old Ubuntu to a newer Ubuntu. 

I some cases, `convert` fails to generate a clipped version of a JPG with clipping path, which is something we have used for years. It just returns a blank image - meaning everything is clipped away.

Looking at the images that fails (with `identify -verbose`) I see that the clipping path data is very large for the image that fails. So one could think that this is an issue with a resource limit, but the server have decent specs:

```
$ convert -list resource
Resource limits:
  Width: 32KP
  Height: 32KP
  List length: 18.446744EP
  Area: 34.36GP
  Memory: 256MiB
  Map: 512MiB
  Disk: 1GiB
  File: 768
  Thread: 12
  Throttle: 0
  Time: unlimited
```

(a lot compared to the older server)

The command I use for testing is this:

`convert input.jpg -alpha transparent -clip -alpha opaque -trim output.png`

**Works**

Input:
![work](https://github.com/ImageMagick/ImageMagick/assets/2204376/bd0ce608-2640-4142-b185-15b035ddbe25)

Output:
<img width="492" alt="Skærmbillede 2023-10-05 kl  13 39 06" src="https://github.com/ImageMagick/ImageMagick/assets/2204376/d5eb37e8-e5b6-4312-a8ac-f5a8ae1c3697">

**Fails**

Input:
![fail](https://github.com/ImageMagick/ImageMagick/assets/2204376/b50b5a85-6772-4adc-a2db-d178fe03efcd)

Output:
<img width="475" alt="Skærmbillede 2023-10-05 kl  13 40 04" src="https://github.com/ImageMagick/ImageMagick/assets/2204376/4064f29f-3adb-45df-bc7e-e6d5219c2c0d">
(image is blank)

Any ideas what could cause this - or how to debug it? Adding `-debug all` gave me a slight headache

**Comment by snibgo** [comment]
Your command works fine for me. The limitations in policy.xml should not be a problem, but your server may restrict memory usage or CPU or something else. Do you get any error messages?

You could capture the output from "-debug all" and post it here.

**Reply by nikolajhave** [reply]
I managed to enable error output and now get this:

`convert-im6.q16: geometry does not contain image 'input.jpg' @ warning/attribute.c/GetImageBoundingBox/247.`

I have attached the complete debug.

[debug-all.txt](https://github.com/ImageMagick/ImageMagick/files/12819087/debug-all.txt)

**Comment by snibgo** [comment, marked as answer]
The first exception in your debug log file is:
```
2023-10-05T15:45:05+02:00 0:00.040 0.230u 6.9.10 Exception convert-im6.q16[987571]: draw.c/RenderMVGContent/3979/Exception
  too many bezier coordinates ` M 1230 1117 C 1228.67 1120.67 1227.33 1124.33 1226 1128 ...
```
So that is the problem.

It works fine for me, with IM v6.9.12-72. Your version is 6.9.10-23. I suggest you upgrade to the current version, 6.9.12-97, at https://legacy.imagemagick.org/script/download.php#windows

Even better, upgrade to v7, if you can. But v7 is not entirely backwards compatible with v6, so you may need to upgrade scripts.

**Reply by nikolajhave** [reply]
Thanks a lot, @snibgo! ... I will try the upgrade. But isn't it weird that this worked in ImageMagick 6.8.9-9?

**Reply by nikolajhave** [reply]
This worked, so once again, thank you for your help! I had to compile a version, since I couldn't make the Redhat-version work with my Ubuntu server.

**Comment by fmw42** [comment]
Are your delegates the same and are they the same versions?

**Comment by snibgo** [comment]
I don't think delegates are used here. But I could be wrong.

> But isn't it weird that this worked in ImageMagick 6.8.9-9?

I don't know. Perhaps 6.8.9-9 ignored bezier coordinates after a certain limit, or accepted them but broke array bounds, or something. And then someone noticed a bug which was fixed but the fix caused an exception when there were too many coordinates, and then that got fixed (perhaps by increasing an array size). But I don't know.

I notice that draw.c contains:
```
#define MaxBezierCoordinates  67108864
```
